// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// Keep in sync with
// http://google3/google/internal/location/nearby/presence/v1/rpcs/certificates.proto
// Messages not used in Chrome have been omitted.

syntax = "proto3";

package ash.nearby.proto;

option optimize_for = LITE_RUNTIME;

import "chromeos/ash/components/nearby/presence/proto/rpc_resources.proto";

// Request for querying the shared credentials for the gaia.
message ListSharedCredentialsRequest {
  // The dusi currently stored on the device.
  string dusi = 1;

  // If it is not set, returns all the shared credentials
  // If it is PRIVATE, returns all the shared credentials from the same gaia.
  IdentityType identity_type = 2;
}

// Response for matched shared credentials.
message ListSharedCredentialsResponse {
  repeated SharedCredential shared_credentials = 1;
}
